### PR TITLE
Move initialization to admin.CreateDomain

### PR DESCRIPTION
**Pull request by gdbelvin**

### CODE REVIEW

File: core/keyserver/epochs_test.go [hunk 1]
```diff
@@ -70,6 +70,15 @@ func TestListMutations(t *testing.T) {
 		t.Fatalf("admin.Write(): %v", err)
 	}
 
+	// Log server
+	numSequencers := 1
+	unused := ""
+	logEnv, err := integration.NewLogEnv(ctx, numSequencers, unused)
```
Comment (pavelkalinnikov) [inline core/keyserver/epochs_test.go]: optionally:
`logEnv, err := integration.NewLogEnv(ctx, 1 /* numSequencers */ , "" /* unused */)`

File: core/adminserver/admin_server.go [hunk 2]
```diff
@@ -83,20 +86,26 @@ var (
...
 	tmap     tpb.TrillianMapClient
 	logAdmin tpb.TrillianAdminClient
 	mapAdmin tpb.TrillianAdminClient
+	domains  domain.Storage
 	keygen   keys.ProtoGenerator
 }
 
 // New returns a KeyTransparencyAdmin implementation.
-func New(domains domain.Storage, tmap tpb.TrillianMapClient, logAdmin, mapAdmin tpb.TrillianAdminClient, keygen keys.ProtoGenerator) *Server {
+func New(tlog tpb.TrillianLogClient,
```
[the remark this answers is not in the document]
Comment (gdbelvin) [reply core/adminserver/admin_server.go]: Looks good

File: core/adminserver/admin_server.go [hunk 3]
```diff
@@ -216,6 +232,50 @@ func (s *Server) CreateDomain(ctx context.Context, in *pb.CreateDomainRequest) (
...
+	}
+	mapRoot, err := s.tmap.GetSignedMapRoot(ctx,
+		&tpb.GetSignedMapRootRequest{MapId: mapID})
+	if err != nil {
+		return fmt.Errorf("GetSignedMapRoot(%v): %v", mapID, err)
+	}
+
+	// If the tree is empty and the map is empty,
+	// add the empty map root to the log.
+	if logRoot.GetSignedLogRoot().GetTreeSize() == 0 &&
```
[the remark this answers is not in the document]
Comment (gdbelvin) [reply core/adminserver/admin_server.go]: Great idea

File: core/keyserver/epochs_test.go (hunk 1 above)
Comment (gdbelvin) [reply core/keyserver/epochs_test.go]: not sure if this is better.

**Pull request merged**